### ISSUE TITLE
Map initially loads as black

### ISSUE THREAD

**Issue by mitsumaui** [TheLastProject/lovelace-valetudo-map-card#4]
Hey!

Thanks for a super awesome HA card, this makes my Valetudo setup much cleaner as it removes the need for MQTT and docker containers.

I did want to raise this to see if it could be improved at all. I noticed on various clients (iOS app, Chrome, Safari) the live map initially loads as black, then resolves into colour after somewhere between 1-3 seconds. It can also occur when resizing the window. 

<img width="460" alt="Screenshot 2019-08-27 at 21 02 47" src="https://user-images.githubusercontent.com/20060563/63804459-1ecdb180-c90f-11e9-9e96-a6d2fd6919a9.png">

It would be great if it could display immediately - depending on where the delay is, or it could initially show the floor colour, before resolving in the objects? Happy to help with any further info you need!

HASS / Lovelace is running on a NUC like device (Intel x5-Z8350 4 x 1.92Ghz) custom built Ubuntu w/ Docker-CE. I see the CPU spike to around 15% when the page loads with the card if that's any help.

**Comment by TheLastProject** [comment]
Yeah I've seen this bug occur on my Firefox too, just no clue why it happens. It must be solvable surely but well, I don't know what causes it so that makes solving it quite tough currently

**Comment by mitsumaui** [comment]
Ha! Lightning fast response! Thanks for taking time to reply. I guess it could be some code optimisation? As it's running client side as far as I can tell. Will see if I can spend some time debugging - however am green with JS :)

**Comment by TheLastProject** [comment]
Looking more into it, the problem is in the way I calculate colours:
```
    // Calculate colours
    const floorColor = this.calculateColor(this, this._config.floor_color, '--valetudo-map-floor-color', '--secondary-background-color');
    const obstacleWeakColor = this.calculateColor(this, this._config.obstacle_weak_color, '--valetudo-map-obstacle-weak-color', '--divider-color');
    const obstacleStrongColor = this.calculateColor(this, this._config.obstacle_strong_color, '--valetudo-map-obstacle-strong-color', '--accent-color');
    const pathColor = this.calculateColor(this, this._config.path_color, '--valetudo-map-path-color', '--primary-text-color');
```

`this` seems like it may be null for a bit, which makes the following call fail:
```
        let possibleColor = getComputedStyle(container).getPropertyValue(color);
```

Thus turning everything black. Everything is already rendered, so it's not about the drawing being slow, it's about the colours not being retrieved initially. Not sure yet why.

**Comment by TheLastProject** [comment]
It's not a true fix, but 9eff10f088cbfac61b236a60bdbf6823f8142d23 at least makes it less visually jarring.
